mount/mount_linux.go

```diff
@@ -106,28 +106,42 @@ func Unmount(target string, flags int) error {
 	return nil
 }
 
-func isFUSE(dir string) (bool, error) {
+func isFUSE(dir string) bool {
 	// fuseSuperMagic is defined in statfs(2)
 	const fuseSuperMagic = 0x65735546
 	var st unix.Statfs_t
 	if err := unix.Statfs(dir, &st); err != nil {
-		return false, err
+		return false
 	}
-	return st.Type == fuseSuperMagic, nil
+	return st.Type == fuseSuperMagic
 }
 
-func unmount(target string, flags int) error {
-	// For FUSE mounts, attempting to execute fusermount helper binary is preferred
-	// https://github.com/containerd/containerd/pull/3765#discussion_r342083514
-	if ok, err := isFUSE(target); err == nil && ok {
-		for _, helperBinary := range []string{"fusermount3", "fusermount"} {
-			cmd := exec.Command(helperBinary, "-u", target)
-			if err := cmd.Run(); err == nil {
-				return nil
-			}
-			// ignore error and try unix.Unmount
+// tryUnmountFUSE checks if target is a FUSE mount, and if so, attempts to unmount
+// using fusermount/fusermount3 helper binary. It returns an error if target is
+// not a FUSE mount, or if unmounting failed.
+//
+// For FUSE mounts, using these helper binaries is preferred, see:
+// https://github.com/containerd/containerd/pull/3765#discussion_r342083514
+func tryUnmountFUSE(target string) error {
+	if !isFUSE(target) {
+		return errors.New("not a FUSE mount")
```

Comment:
Do we want to include the `target` in the message?

Reply:
Oh! I see I didn't reply to you; I left it out mostly because this error was more to indicate we skipped fuse-unmounting, in which case a regular unmount would be performed (and give the usual errors if that would fail).

But this error is now removed in my latest change, so this should be resolved 😅